internal/fs/implicit_dirs_with_cache_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where we do not attempt to write to
+// the file system at all. Rather we set up contents in a GCS bucket out of
+// band, wait for them to be available, and then read them via the file system.
+
+package fs_test
+
+import (
+	"os/exec"
+	"path"
+	"time"
+
+	. "github.com/jacobsa/ogletest"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type ImplicitDirsWithCacheTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&ImplicitDirsWithCacheTest{})
+}
+
+func (t *ImplicitDirsWithCacheTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.DirTypeCacheTTL = time.Minute * 1
+	t.fsTest.SetUpTestSuite()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *ImplicitDirsWithCacheTest) TestRemoveAll() {
+
+	/* When rm -r or os.RemoveAll call is made, the following calls are made in order
+		 1. unlink all nested files,
+		 2. lookupInode call on implicit directory
+		 3. Rmdir on the directory.
+
+	When type cache ttl is set, we construct an implicitDir even though one doesn't
+	exist on GCS (https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L452),
+	and thus, we get rmDir call to GCSFuse.
+	Whereas when ttl is zero, lookupInode call itself fails because object is
+	not present in GCS.
+	*/
+
+	// Create an implicit directory.
+	AssertEq(
+		nil,
+		t.createObjects(
+			map[string]string{
+				// File
+				"foo/bar": "",
+			}))
+
+	//err := os.RemoveAll(path.Join(mntDir, "foo"))
```

Comment:
please uncomment this and comment cmd.

Reply:
okay, please correct the comment in this case (this comment: "When rm -r or os.RemoveAll call is made, the following calls are made in order")